diff --git a/build/rive_build_config.lua b/build/rive_build_config.lua
index f6ff5103..5b69a75d 100644
--- a/build/rive_build_config.lua
+++ b/build/rive_build_config.lua
@@ -325,26 +325,26 @@ if _OPTIONS['os'] == 'android' then
     filter('options:arch=x86')
     do
         architecture('x86')
-        buildoptions({ '--target=i686-none-linux-android21' })
-        linkoptions({ '--target=i686-none-linux-android21' })
+        buildoptions({ '--target=i686-none-linux-android31' })
+        linkoptions({ '--target=i686-none-linux-android31' })
     end
     filter('options:arch=x64')
     do
         architecture('x64')
-        buildoptions({ '--target=x86_64-none-linux-android21' })
-        linkoptions({ '--target=x86_64-none-linux-android21' })
+        buildoptions({ '--target=x86_64-none-linux-android31' })
+        linkoptions({ '--target=x86_64-none-linux-android31' })
     end
     filter('options:arch=arm')
     do
         architecture('arm')
-        buildoptions({ '--target=armv7a-none-linux-android21' })
-        linkoptions({ '--target=armv7a-none-linux-android21' })
+        buildoptions({ '--target=armv7a-none-linux-android31' })
+        linkoptions({ '--target=armv7a-none-linux-android31' })
     end
     filter('options:arch=arm64')
     do
         architecture('arm64')
-        buildoptions({ '--target=aarch64-none-linux-android21' })
-        linkoptions({ '--target=aarch64-none-linux-android21' })
+        buildoptions({ '--target=aarch64-none-linux-android31' })
+        linkoptions({ '--target=aarch64-none-linux-android31' })
     end
 
     filter({})
diff --git a/build/setup_compiler.lua b/build/setup_compiler.lua
index 88f54630..5dec770a 100644
--- a/build/setup_compiler.lua
+++ b/build/setup_compiler.lua
@@ -212,26 +212,26 @@ if _OPTIONS['os'] == 'android' then
     filter('options:arch=x86')
     do
         architecture('x86')
-        buildoptions({ '--target=i686-none-linux-android21' })
-        linkoptions({ '--target=i686-none-linux-android21' })
+        buildoptions({ '--target=i686-none-linux-android31' })
+        linkoptions({ '--target=i686-none-linux-android31' })
     end
     filter('options:arch=x64')
     do
         architecture('x64')
-        buildoptions({ '--target=x86_64-none-linux-android21' })
-        linkoptions({ '--target=x86_64-none-linux-android21' })
+        buildoptions({ '--target=x86_64-none-linux-android31' })
+        linkoptions({ '--target=x86_64-none-linux-android31' })
     end
     filter('options:arch=arm')
     do
         architecture('arm')
-        buildoptions({ '--target=armv7a-none-linux-android21' })
-        linkoptions({ '--target=armv7a-none-linux-android21' })
+        buildoptions({ '--target=armv7a-none-linux-android31' })
+        linkoptions({ '--target=armv7a-none-linux-android31' })
     end
     filter('options:arch=arm64')
     do
         architecture('arm64')
-        buildoptions({ '--target=aarch64-none-linux-android21' })
-        linkoptions({ '--target=aarch64-none-linux-android21' })
+        buildoptions({ '--target=aarch64-none-linux-android31' })
+        linkoptions({ '--target=aarch64-none-linux-android31' })
     end
     filter({})
 end
diff --git a/dependencies/premake5_harfbuzz_v2.lua b/dependencies/premake5_harfbuzz_v2.lua
index 428aa156..27b12025 100644
--- a/dependencies/premake5_harfbuzz_v2.lua
+++ b/dependencies/premake5_harfbuzz_v2.lua
@@ -266,7 +266,7 @@ do
 
     filter('options:config=release')
     do
-        optimize('Size')
+        -- optimize('Size')
     end
 
     filter({ 'options:not no-harfbuzz-renames' })
diff --git a/dependencies/premake5_sheenbidi_v2.lua b/dependencies/premake5_sheenbidi_v2.lua
index 891c963e..d9019afd 100644
--- a/dependencies/premake5_sheenbidi_v2.lua
+++ b/dependencies/premake5_sheenbidi_v2.lua
@@ -47,7 +47,7 @@ do
     filter('options:config=release')
     do
         defines({ 'SB_CONFIG_UNITY' })
-        optimize('Size')
+        -- optimize('Size')
     end
 
     filter({ 'system:linux' })

diff --git a/premake5_v2.lua b/premake5_v2.lua
index 3d6c8aa5..d5078a55 100644
--- a/premake5_v2.lua
+++ b/premake5_v2.lua
@@ -50,7 +50,7 @@ do
 
     files({ 'src/**.cpp' })
 
-    flags({ 'FatalCompileWarnings' })
+    -- flags({ 'FatalCompileWarnings' })
 
     filter({ 'options:with_rive_text', 'options:not no-harfbuzz-renames' })
     do
